### PR TITLE
Add performance benchmarks for helper functions.

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Add performance benchmarks for helper functions.

Resolves: #461 

Signed-off-by: Alan Jowett <alanjo@microsoft.com>

### CODE REVIEW

File: tests/performance/performance.cpp [hunk 1]
```diff
@@ -233,8 +260,59 @@ typedef class _ebpf_program_test_state
...
+        ebpf_epoch_exit();
+    }
+
+    void
+    test_find_write()
+    {
+        uint32_t key = 0;
+        uint64_t* value = nullptr;
+        ebpf_epoch_enter();
+        ebpf_map_find_entry(map, 0, (uint8_t*)&key, 0, (uint8_t*)&value, EBPF_MAP_FLAG_HELPER);
```
Comment (dthaler) [inline tests/performance/performance.cpp:294]: this looks identical to line 284.   I suspect having two identical functions with different names (read vs write) is not what you intended.

Comment (Alan-Jowett) [reply tests/performance/performance.cpp:302]: Good catch.

**Comment by Alan-Jowett** [comment]
Sample output:
```
Filters: test_bpf*
test_bpf_get_prandom_u32,1,2280
test_bpf_get_prandom_u32,0,2234
test_bpf_ktime_get_boot_ns,1,164
test_bpf_ktime_get_boot_ns,0,80
test_bpf_get_smp_processor_id,1,135
test_bpf_get_smp_processor_id,0,75
test_bpf_map_lookup_elem_read<BPF_MAP_TYPE_HASH>,1,174
test_bpf_map_lookup_elem_read<BPF_MAP_TYPE_HASH>,0,112
test_bpf_map_lookup_elem_read<BPF_MAP_TYPE_ARRAY>,1,342
test_bpf_map_lookup_elem_read<BPF_MAP_TYPE_ARRAY>,0,71
test_bpf_map_lookup_elem_read<BPF_MAP_TYPE_PERCPU_HASH>,1,188
test_bpf_map_lookup_elem_read<BPF_MAP_TYPE_PERCPU_HASH>,0,322
test_bpf_map_lookup_elem_read<BPF_MAP_TYPE_PERCPU_ARRAY>,1,381
test_bpf_map_lookup_elem_read<BPF_MAP_TYPE_PERCPU_ARRAY>,0,100
test_bpf_map_lookup_elem_write<BPF_MAP_TYPE_HASH>,1,403
test_bpf_map_lookup_elem_write<BPF_MAP_TYPE_HASH>,0,358
test_bpf_map_lookup_elem_write<BPF_MAP_TYPE_ARRAY>,1,450
test_bpf_map_lookup_elem_write<BPF_MAP_TYPE_ARRAY>,0,473
test_bpf_map_lookup_elem_write<BPF_MAP_TYPE_PERCPU_HASH>,1,278
test_bpf_map_lookup_elem_write<BPF_MAP_TYPE_PERCPU_HASH>,0,117
test_bpf_map_lookup_elem_write<BPF_MAP_TYPE_PERCPU_ARRAY>,1,239
test_bpf_map_lookup_elem_write<BPF_MAP_TYPE_PERCPU_ARRAY>,0,337
test_bpf_map_update_elem<BPF_MAP_TYPE_HASH>,1,2394
test_bpf_map_update_elem<BPF_MAP_TYPE_HASH>,0,2329
test_bpf_map_update_elem<BPF_MAP_TYPE_ARRAY>,1,295
test_bpf_map_update_elem<BPF_MAP_TYPE_ARRAY>,0,461
test_bpf_map_update_elem<BPF_MAP_TYPE_PERCPU_HASH>,1,305
test_bpf_map_update_elem<BPF_MAP_TYPE_PERCPU_HASH>,0,298
test_bpf_map_update_elem<BPF_MAP_TYPE_PERCPU_ARRAY>,1,313
test_bpf_map_update_elem<BPF_MAP_TYPE_PERCPU_ARRAY>,0,179
```

**Review by dthaler — APPROVED** [review]

File: tests/performance/performance.cpp [hunk 2]
```diff
@@ -266,6 +353,124 @@ _ebpf_program_invoke()
...
+    name += ">";
+    _performance_measure measure(name.c_str(), preemptible, _map_find_read_test, iterations);
+    measure.run_test();
+}
+
+template <ebpf_map_type_t map_type>
+void
+test_bpf_map_lookup_elem_write(bool preemptible)
+{
+    size_t iterations = PERFORMANCE_MEASURE_ITERATION_COUNT;
```
Comment (saxena-anurag) [inline tests/performance/performance.cpp:418]: Do we need to call `ebpf_core_initiate()` in this test?

Comment (Alan-Jowett) [reply tests/performance/performance.cpp:418]: Yes, and it currently does via the _test_helper_end_to_end constructor/destructor.

**Review by saxena-anurag — APPROVED** [review]

**Pull request merged**